Tests failing with numba 0.55.0

The new [0.55.0 release of Numba](https://numba.readthedocs.io/en/stable/release-notes.html#version-0-55-0-13-january-2021) supports NumPy 1.21, but a test is failing:

```
=================================== FAILURES ===================================
__________________________ test_rogers_huff_r_between __________________________

    def test_rogers_huff_r_between():
        gna = np.array([[0, 1, 2]])
        gnb = np.array([[0, 1, 2]])
        npt.assert_allclose(rogers_huff_r_between(gna[0], gnb[0]), 1.0, rtol=1e-06)
        npt.assert_allclose(rogers_huff_r2_between(gna[0], gnb[0]), 1.0, rtol=1e-06)
        npt.assert_allclose(
            allel.rogers_huff_r_between(gna, gnb),
            rogers_huff_r_between(gna[0], gnb[0]),
            rtol=1e-06,
        )
    
        gna = np.array([[0, 1, 2]])
        gnb = np.array([[2, 1, 0]])
        npt.assert_allclose(rogers_huff_r_between(gna[0], gnb[0]), -1.0, rtol=1e-06)
        npt.assert_allclose(rogers_huff_r2_between(gna[0], gnb[0]), 1.0, rtol=1e-06)
        npt.assert_allclose(
            allel.rogers_huff_r_between(gna, gnb),
            rogers_huff_r_between(gna[0], gnb[0]),
            rtol=1e-06,
        )
    
        gna = np.array([[0, 0, 0]])
        gnb = np.array([[1, 1, 1]])
        assert np.isnan(rogers_huff_r_between(gna[0], gnb[0]))
        assert np.isnan(rogers_huff_r2_between(gna[0], gnb[0]))
        assert np.isnan(allel.rogers_huff_r_between(gna, gnb))
    
        # a case where scikit-allel is different due to its use of float32
        gna = np.full((1, 49), 2)
        gnb = np.full((1, 49), 2)
>       npt.assert_allclose(rogers_huff_r_between(gna[0], gnb[0]), 1.0, rtol=1e-06)
E       AssertionError: 
E       Not equal to tolerance rtol=1e-06, atol=0
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 0.11654779
E       Max relative difference: 0.11654779
E        x: array(0.883452)
E        y: array(1.)
```

It was failing in Numba 0.54 too, but we have excluded that version due to incompatibility with later NumPy releases. We should work out what the underlying issue is and fix that.

--- a/requirements.txt
+++ b/requirements.txt
@@ -5,6 +5,6 @@ distributed >= 2021.10.0
 dask-ml
 scipy
 typing-extensions
-numba != 0.54.*
+numba
 zarr
 fsspec != 2021.6.*

--- a/setup.cfg
+++ b/setup.cfg
@@ -33,7 +33,7 @@ install_requires =
     dask-ml
     scipy
     zarr
-    numba != 0.54.*
+    numba
     typing-extensions
     fsspec != 2021.6.*
     setuptools >= 41.2  # For pkg_resources

--- a/sgkit/stats/ld.py
+++ b/sgkit/stats/ld.py
@@ -15,7 +15,7 @@ from sgkit.typing import ArrayLike, DType
 from sgkit.window import _get_chunked_windows, _sizes_to_start_offsets, has_windows
 
 
-@njit(nogil=True, fastmath=True, cache=True)  # type: ignore
+@njit(nogil=True, fastmath=False, cache=True)  # type: ignore
 def rogers_huff_r_between(gn0: ArrayLike, gn1: ArrayLike) -> float:  # pragma: no cover
     """Rogers Huff *r*.
 

--- a/sgkit/tests/test_ld.py
+++ b/sgkit/tests/test_ld.py
@@ -49,11 +49,17 @@ def test_rogers_huff_r_between():
     assert np.isnan(rogers_huff_r2_between(gna[0], gnb[0]))
     assert np.isnan(allel.rogers_huff_r_between(gna, gnb))
 
-    # a case where scikit-allel is different due to its use of float32
+    gna = np.array([[1, 1, 1]])
+    gnb = np.array([[1, 1, 1]])
+    assert np.isnan(rogers_huff_r_between(gna[0], gnb[0]))
+    assert np.isnan(rogers_huff_r2_between(gna[0], gnb[0]))
+    assert np.isnan(allel.rogers_huff_r_between(gna, gnb))
+
+    # a case which fails if fastmath=True is enabled for rogers_huff_r_between
     gna = np.full((1, 49), 2)
     gnb = np.full((1, 49), 2)
-    npt.assert_allclose(rogers_huff_r_between(gna[0], gnb[0]), 1.0, rtol=1e-06)
-    npt.assert_allclose(rogers_huff_r2_between(gna[0], gnb[0]), 1.0, rtol=1e-06)
+    assert np.isnan(rogers_huff_r_between(gna[0], gnb[0]))
+    assert np.isnan(rogers_huff_r2_between(gna[0], gnb[0]))
     assert np.isnan(allel.rogers_huff_r_between(gna, gnb))
 
 

